test/arrayops.jl

```diff
@@ -2079,6 +2079,34 @@ end
     @test size(a) == size(b)
 end
 
+@testset "type constructor Array{T, N}(nothing, d...) works (especially for N>3)" for T in (Int, String),
+                                                                                      U in (Void, Missing)
+    a = Array{Union{T, U}}(U(), 10)
+    b = Vector{Union{T, U}}(U(), 10)
+    @test size(a) == (10,)
+    @test size(a, 1) == 10
+    @test size(a,2,1) == (1,10)
+    @test size(a) == size(b)
+    @test all(x -> x isa U, a)
+    @test all(x -> x isa U, b)
+    a = Array{Union{T, U}}(U(), 2,3)
+    b = Matrix{Union{T, U}}(U(), 2,3)
+    @test size(a) == (2,3)
+    @test size(a,4,3,2,1) == (1,1,3,2)
+    @test size(a,1,2) == (2,3)
+    @test size(a) == size(b)
+    @test all(x -> x isa U, a)
+    @test all(x -> x isa U, b)
+    a = Array{Union{T, U}}(U(), 9,8,7,6,5,4,3,2,1)
+    b = Array{Union{T, U},9}(U(), 9,8,7,6,5,4,3,2,1)
+    @test size(a,1,1) == (9,9)
+    @test size(a,4) == 6
+    @test size(a,9,8,7,6,5,4,3,2,19,8,7,6,5,4,3,2,1) == (1,2,3,4,5,6,7,8,1,2,3,4,5,6,7,8,9)
+    @test size(a) == size(b)
+    @test all(x -> x isa U, a)
+    @test all(x -> x isa U, b)
+end
```

Comment:
Why all the tests of `size`? They seem somewhat unrelated :).

Reply:
Indeed. I just copied the tests used for the standard constructors without noticing these extra lines.